test/ProgressOnderwijsUtils.Tests/RootedTreeTest.cs

```diff
@@ -47,5 +47,26 @@ public void ReplaceSubTree_can_replace_nested_node()
                     )
                 ).RootHere()));
         }
+
+        [Fact]
+        public void RootedTree_Sibling_Implementation_Works_Correctly()
+        {
+            var tree =
+                Tree.Node(1,
+                    Tree.Node(2),
+                    Tree.Node(3,
+                        Tree.Node(4),
+                        Tree.Node(5)
+                    ),
+                    Tree.Node(6)
+                ).RootHere();
+
+            PAssert.That(() => !tree.PreviousSibling().HasValue);
+            PAssert.That(() => !tree.NextSibling().HasValue);
+            PAssert.That(() => !tree.Children[0].PreviousSibling().HasValue);
+            PAssert.That(() => tree.Children[0].NextSibling().HasValue);
+            PAssert.That(() => tree.Children[0].NextSibling().AssertNotNull().NextSibling().HasValue);
+            PAssert.That(() => !tree.Children[0].NextSibling().AssertNotNull().NextSibling().AssertNotNull().NextSibling().HasValue);

```

Comment:
```C#
PAssert.That(() => !tree.Parent.PreviousSibling().HasValue);
PAssert.That(() => !tree.Parent.NextSibling().HasValue);
```

Reply:
Terechte opmerking, maar _on a sidenote_ is t terecht dat `IsRoot` aanroepen crashed indien er geen parent is?